pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -170,8 +170,22 @@ in stdenv.mkDerivation {
     "-Dman=true"
   ];
 
-  preConfigure = ''
+  preConfigure = let
+    defaultDnsServers = [
+      # We use these public name services, ordered by their privacy policy (hopefully):
+      #  * Cloudflare (https://1.1.1.1/)
+      #  * Quad9 without filtering (https://www.quad9.net/)
+      #  * Google (https://developers.google.com/speed/public-dns/)
+      "1.1.1.1"
+      "9.9.9.10"
+      "8.8.8.8"
+      "2606:4700:4700::1111"
+      "2620:fe::10"
+      "2001:4860:4860::8888"
+    ];
+  in ''
     mesonFlagsArray+=(-Dntp-servers="0.nixos.pool.ntp.org 1.nixos.pool.ntp.org 2.nixos.pool.ntp.org 3.nixos.pool.ntp.org")
+    mesonFlagsArray+=("-Ddns-servers=${toString defaultDnsServers}")
```

Comment:
It does not change much actually... But it is probably good to have had a discussion about it.
By default nixos does not use systemd-resolved btw.